libr/util/list.c

```diff
@@ -414,33 +472,113 @@ R_API void *r_list_get_by_string(const RList *list, int off, const char *str) {
 	char *p;
 	RListIter *iter;
 	r_list_foreach (list, iter, p) {
-		const char *ptr = p+off;
-		if (!strcmp (str, ptr))
+		const char *ptr = p + off;
+		if (!strcmp (str, ptr)) {
 			return p;
+		}
 	}
 	return NULL;
 }
 
-R_API RListIter *r_list_contains (const RList *list, const void *p) {
+R_API RListIter *r_list_contains(const RList *list, const void *p) {
 	void *q;
 	RListIter *iter;
 	r_list_foreach (list, iter, q) {
-		if (p == q)
+		if (p == q) {
 			return iter;
+		}
 	}
 	return NULL;
 }
 
-R_API RListIter *r_list_find (const RList *list, const void *p, RListComparator cmp) {
+R_API RListIter *r_list_find(const RList *list, const void *p, RListComparator cmp) {
 	void *q;
 	RListIter *iter;
 	r_list_foreach (list, iter, q) {
-		if (cmp (p, q) == 0)
+		if (!cmp (p, q)) {
 			return iter;
+		}
 	}
 	return NULL;
 }
 
+
+RListIter *_sortedmerge(RListIter *first, RListIter *second, RListComparator cmp) {
+	RListIter *result = NULL;
+	if (!first) { 
+		return second;
+	}
+	if (!second) {
+		return first;
+	}
+	if (cmp (first->data, second->data) < 0) {
+		result = first;
+		result->n = _sortedmerge (first->n, second, cmp);
+		result->n->p = result;
+		result->p = NULL;
+	} else {
+		result = second;
+		result->n = _sortedmerge (first, second->n, cmp);
+		result->n->p = result;
+		result->p = NULL;
+	}
+	return result;
+}
+
+
+void _r_list_half_split(RListIter *head, RListIter **first, RListIter **second) {
+	RListIter *fast;
+	RListIter *slow;
+	RListIter *ref = head;
+	int count = 0, middle;
+
+	if (!head || !head->n) {
+		*first = NULL;
+		*second = NULL;
+		return;
+	} else {
+		while (ref) {
+			count++;
+			ref = ref->n;
+		}
+		middle = count / 2;
+		slow = head;
+		fast = slow->n;
+		while (fast && middle-- > 0) {
+			fast = fast->n;
+			if (fast) {
+				slow = slow->n;
+				fast = fast->n;
+			}
+		}
+		*first = head;
+		*second = slow->n;
+		slow->n = NULL;
+	}
+}
+
+void _merge_sort(RListIter **head_ref, RListComparator cmp) {
```

Comment:
this and others should be `static`. Also, i'm not sure we usually use `_function_name` syntax